clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -513,10 +463,10 @@ private void handleListOffsetResponse(TopicPartition topicPartition,
 
     private Set<TopicPartition> fetchablePartitions() {
         Set<TopicPartition> fetchable = subscriptions.fetchablePartitions();
-        if (records.isEmpty())
-            return fetchable;
-        for (PartitionRecords<K, V> partitionRecords : records)
-            fetchable.remove(partitionRecords.partition);
+        if (nextInLineRecords != null && !nextInLineRecords.isConsumed())
```

Comment:
I debated on this. I agree that the redundancy is slightly annoying, but it also seems to make the logic in `fetchedRecords` more self-contained. What do you think?